File: SolutionsCSharp/TwoSum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeWarsSolutions
{
    internal class TwoSum
    {
        Dictionary<int, int> myDictionary = new Dictionary<int,int>();

        public static int[] TwoSumMethod(int[] nums, int target)
        {
            Dictionary<int, int> myDictionary = new Dictionary<int, int>();

            for (int i = 0; i < nums.Length; i++)
            {
                if (myDictionary.ContainsKey(nums[i]))
                {
                    int[] result = { myDictionary[nums[i]], i };
                    return result;
                }
                myDictionary[target - nums[i]] = i;
            }

            int[] resultFail = { 0, 0 };
            return resultFail;
        }
    }
}

File: solutions_python/two_sum.py
def two_sum(nums, target):
    my_map = {}
    count = 0
    for num in nums:
        diff = target - num
        if num in my_map:
            index1 = my_map[num]
            index2 = count
        else:
            my_map[diff] = count
        count += 1
    result = [index1, index2]
    return result